File: data/etl/dependencies/connect_db.py
import sqlite3

### CONNECT (OR CREATE) DATABASE
def connect_db(database_name, database_path):
    try:
        with sqlite3.connect(database_path) as conn:
            print(f"Opened SQLite database {database_name} with SQLite version {sqlite3.sqlite_version} successfully.")
            return sqlite3.connect(database_path)
        
    except sqlite3.Error as e:
        print(f"Failed to open database {database_name}:\n", e)
        
if __name__ == "__main__":
    database_name = "mls_stats.db"
    database_path = f"data/database/{database_name}"
    connect_db(database_name, database_path)

File: app-duels-mapping/connect_db.js
const fs = require("fs");
const path = require("path");
const sqlite3 = require("sqlite3").verbose();

// Load and parse the config JSON
const configPath = path.join(__dirname, "public", "data", "data_vars.json");
const config = JSON.parse(fs.readFileSync(configPath, "utf-8"));

// Extract database path and name
const databaseDir = config.database.path;
const databaseName = config.database.name;

// Build the full absolute path to the database
const fullDbPath = path.join(__dirname, databaseDir, databaseName);

// Connect to SQLite database in read-only mode
const db = new sqlite3.Database(fullDbPath, sqlite3.OPEN_READONLY, (err) => {
  if (err) {
    return console.error("❌ Error opening database:", err.message);
  } else {
    console.log(`✅ Opened SQLite database ${databaseName} successfully.`);
  }
});

module.exports = db;
